packages/core/src/QueryBuilder/QueryBuilder.tsx

```diff
@@ -48,9 +49,18 @@ export interface HvQueryBuilderProps {
   labels?: HvQueryBuilderLabels;
   /** Whether the query builder is in read-only mode. */
   readOnly?: boolean;
-  /** Renderers for custom attribute types. */
+  /**
+   * Operators that should use the empty value renderer when selected.
+   *
+   * When one of the listed operators is selected, the rule value is reset and an empty component is rendered.
+   * This property takes priority over `renderers`.
+   *
+   * @default ["Empty", "IsNotEmpty"]
+   * */
+  emptyRenderer?: string[];
```

Comment:
(Just a small note - I don't have a better suggestion, this is a good compromise)

In one hand, this adds an extra layer of complexity when dealing with/learning about QueryBuilder renderers, and will likely mean we'll be maintaining this in the next major versions.
But but on the other hand, it allows opting-out of the weird hard-coded `"Empty"` and `"IsNotEmpty"` values.

We also can't make nullish renderers the `EmptyValue` because that'd be a breaking change (as our default is `TextValue`)

Reply:
Yes, I was also on the fence about adding or not this property but, in the end, I decided to add it mainly because of the breaking changes it would introduce not having it: users that are already using `renderers` may be expecting the "magic" behaviour we have with `Empty` and `IsNotEmpty` 😅